.github/workflows/test-coverage.yml

```diff
@@ -0,0 +1,107 @@
+---
+name: sonarcloud
+
+on:
+  push:
+    branches: [main, develop]
+  pull_request:
+    branches: [main, develop]
+  pull_request_target:
```

Comment:
Is it necessary to allow `pull_request_target` in this workflow? I think we should avoid it.